language/tools/vm-genesis/src/lib.rs

```diff
@@ -0,0 +1,597 @@
+        // TODO: publish module first so we don't need this call
+        runtime.cache_module(module.clone());
+    }
+
+    // create execution data for the runtime
+    let state_view = FakeStateView;
+    let gas_schedule = CostTable::zero();
+    let data_cache = BlockDataCache::new(&state_view);
+    // the interpreter context will contain the WriteSet after execution
+    let mut interpreter_context =
+        TransactionExecutionContext::new(GasUnits::new(100_000_000), &data_cache);
+
+    // generate the genesis WriteSet
+    let genesis_write_set = {
+        {
+            create_and_initialize_main_accounts(
+                &runtime,
+                &gas_schedule,
+                &mut interpreter_context,
+                &public_key,
+                initial_gas_schedule(&runtime, &data_cache),
+            );
+            create_and_initialize_validator_and_discovery_set(
+                &runtime,
+                &gas_schedule,
+                &mut interpreter_context,
+                &validator_set,
+                &discovery_set,
+            );
+            reconfigure(&runtime, &gas_schedule, &mut interpreter_context);
+            publish_stdlib(&mut interpreter_context, modules);
+
+            verify_genesis_write_set(interpreter_context.events(), &validator_set, &discovery_set);
+            ChangeSet::new(
+                interpreter_context
+                    .make_write_set()
+                    .expect("Genesis WriteSet failure"),
+                interpreter_context.events().to_vec(),
+            )
+        }
+    };
+    let transaction =
+        RawTransaction::new_change_set(account_config::association_address(), 0, genesis_write_set);
+    transaction.sign(private_key, public_key).unwrap()
+}
+
+/// Create an initialize Association, Transaction Fee and Core Code accounts.
+fn create_and_initialize_main_accounts(
+    runtime: &VMRuntime,
+    gas_schedule: &CostTable,
+    interpreter_context: &mut TransactionExecutionContext,
+    public_key: &Ed25519PublicKey,
+    initial_gas_schedule: Value,
+) {
+    const INIT_BALANCE: u64 = 1_000_000_000;
+
+    let association_addr = account_config::association_address();
+    let mut txn_data = TransactionMetadata::default();
+    txn_data.sender = association_addr;
+
+    // create the association account
+    runtime
+        .create_account(
+            interpreter_context,
+            &txn_data,
+            &gas_schedule,
+            association_addr,
+        )
+        .unwrap_or_else(|_| {
+            panic!(
+                "Failure creating association account {:?}",
+                association_addr
+            )
+        });
+
+    // create the transaction fee account
+    let transaction_fee_address = account_config::transaction_fee_address();
+    runtime
+        .create_account(
+            interpreter_context,
+            &txn_data,
+            &gas_schedule,
+            transaction_fee_address,
+        )
```

Comment:
This should be equivalent to expect?

Reply:
not sure what you mean here and I assume you are referring to `unwrap_or_else`?
If so yes it's equivalent if not for the fact that by using `format!` clippy complains that we should use `unwrap_or_else`... go figure....